x/concentrated-liquidity/lp.go

```diff
@@ -250,6 +250,57 @@ func (k Keeper) withdrawPosition(ctx sdk.Context, owner sdk.AccAddress, position
 	return actualAmount0.Neg(), actualAmount1.Neg(), nil
 }
 
+// nolint: unused
+// addToPosition attempts to add liquidityAmount to a position with the given pool id in the given tick range.
+// Returns error if
+// - Withdrawing full position fails
+// - Creating new position with added liquidity fails
+// - Position is superfluid staked
+// TODO: handle adding to SFS positions
+func (k Keeper) addToPosition(ctx sdk.Context, owner sdk.AccAddress, positionId uint64, amount0Added, amount1Added sdk.Int) (uint64, sdk.Int, sdk.Int, error) {
+	position, err := k.GetPosition(ctx, positionId)
+	if err != nil {
+		return 0, sdk.Int{}, sdk.Int{}, err
+	}
+
+	if amount0Added.IsNegative() || amount1Added.IsNegative() {
+		return 0, sdk.Int{}, sdk.Int{}, types.NegativeAmountAddedError{PositionId: position.PositionId, Asset0Amount: amount0Added, Asset1Amount: amount1Added}
+	}
+
+	// If the position is superfluid staked, return error.
+	// TODO: handle this case to allow LPs to add to SFS positions
+	positionHasUnderlyingLock, err := k.positionHasUnderlyingLockInState(ctx, positionId)
+	if err != nil {
+		return 0, sdk.Int{}, sdk.Int{}, err
+	}
+	if positionHasUnderlyingLock {
+		return 0, sdk.Int{}, sdk.Int{}, types.PositionSuperfluidStakedError{PositionId: position.PositionId}
+	}
+
+	// Withdraw full position.
+	amount0Withdrawn, amount1Withdrawn, err := k.withdrawPosition(ctx, owner, positionId, position.Liquidity)
+	if err != nil {
+		return 0, sdk.Int{}, sdk.Int{}, err
+	}
+
+	anyPositionsRemainingInPool, err := k.hasAnyPositionForPool(ctx, position.PoolId)
+	if err != nil {
+		return 0, sdk.Int{}, sdk.Int{}, err
+	}
+
+	if !anyPositionsRemainingInPool {
+		return 0, sdk.Int{}, sdk.Int{}, types.AddToLastPositionInPoolError{PoolId: position.PoolId, PositionId: position.PositionId}
+	}
+
+	// Create new position with updated liquidity.
+	newPositionId, actualAmount0, actualAmount1, _, _, err := k.createPosition(ctx, position.PoolId, owner, amount0Withdrawn.Add(amount0Added), amount1Withdrawn.Add(amount1Added), sdk.ZeroInt(), sdk.ZeroInt(), position.LowerTick, position.UpperTick)
```

Comment:
imo, the min for amount0 and amount1 can be set to the previous amount0 and amount1 withdrawn